lib/topological_inventory/sync/host_inventory_response_worker.rb

```diff
@@ -0,0 +1,127 @@
+require 'base64'
+require "concurrent"
+require 'stringio'
+require 'zlib'
+
+module TopologicalInventory
+  class Sync
+    class ResponseWorker
+      attr_reader :started
+      alias started? started
+
+      def initialize(config, logger)
+        self.config              = config
+        self.lock                = Mutex.new
+        self.logger              = logger
+        self.started             = Concurrent::AtomicBoolean.new(false)
+        self.registered_messages = Concurrent::Map.new
+      end
+
+      # Start listening on Kafka
+      def start
+        lock.synchronize do
+          return if started.value
+
+          started.value = true
+          self.listener = Thread.new { listen while started.value }
```

Comment:
maintenance thread is needed here to clean up registered messages (and log it as an error) after some timeout (let's say 1 hour). In case of some responses won't come the `registered_mesages` map will grow until end of pod life